tests/benchmarks/test_rechunk.py

```diff
@@ -35,47 +37,73 @@ def configure_rechunking(request, memory_multiplier):
             yield
 
 
-@pytest.fixture(params=["8 MiB", "128 MiB"])
-def configure_chunksize(request, memory_multiplier):
-    if memory_multiplier > 0.4 and parse_bytes(request.param) < parse_bytes("64 MiB"):
-        pytest.skip("too slow")
-
-    with dask.config.set({"array.chunk-size": request.param}):
-        yield
+@pytest.fixture(
+    params=[
+        pytest.param("tasks", marks=pytest.mark.shuffle_tasks),
+        pytest.param("p2p", marks=[pytest.mark.shuffle_p2p, requires_p2p_rechunk]),
+    ]
+)
+def configure_rechunking_out_of_core(request):
+    if request.param == "tasks":
+        with dask.config.set({"array.rechunk.method": "tasks"}):
+            yield
+    else:
+        with dask.config.set(
+            {
+                "array.rechunk.method": "p2p",
+                "distributed.p2p.disk": True,
+            }
+        ):
+            yield
 
 
 def test_tiles_to_rows(
     # Order matters: don't initialize client when skipping test
-    memory_multiplier,
-    configure_chunksize,
-    configure_rechunking,
+    input_chunk_size,
+    configure_rechunking_in_memory,
     small_client,
 ):
     """2D array sliced into square tiles becomes sliced by columns.
     This use case can be broken down into N independent problems.
     In task rechunk, this generates O(N) intermediate tasks and graph edges.
     """
     memory = cluster_memory(small_client)
-    shape = scaled_array_shape(memory * memory_multiplier, ("x", "x"))
+    shape = scaled_array_shape(memory * 1.5, ("x", "x"))
 
-    a = da.random.random(shape, chunks="auto")
+    a = da.random.random(shape, chunks=input_chunk_size)
     a = a.rechunk((-1, "auto")).sum()
     wait(a, small_client, timeout=600)
 
 
-def test_swap_axes(
+def test_swap_axes_in_memory(
```

Comment:
I've split `test_swap_axes` into different tests to make it easier to read the charts.